controller/telemetry/server.go

```diff
@@ -184,29 +184,52 @@ func NewServer(addr, prometheusUrl string, ignoredNamespaces []string, kubeconfi
 func (s *server) Query(ctx context.Context, req *read.QueryRequest) (*read.QueryResponse, error) {
 	log.Debugf("Query request: %+v", req)
 
-	start := time.Unix(0, req.StartMs*int64(time.Millisecond))
-	end := time.Unix(0, req.EndMs*int64(time.Millisecond))
+	var res model.Value
+	var err error
 
-	step, err := time.ParseDuration(req.Step)
-	if err != nil {
-		log.Errorf("ParseDuration(%+v) failed with: %+v", req.Step, err)
-		return nil, err
-	}
+	if req.StartMs != 0 && req.EndMs != 0 && req.Step != "" {
+		start := time.Unix(0, req.StartMs*int64(time.Millisecond))
+		end := time.Unix(0, req.EndMs*int64(time.Millisecond))
+		step, err := time.ParseDuration(req.Step)
+		if err != nil {
+			log.Errorf("ParseDuration(%+v) failed with: %+v", req.Step, err)
+			return nil, err
+		}
 
-	queryRange := v1.Range{Start: start, End: end, Step: step}
-	res, err := s.prometheusApi.QueryRange(ctx, req.Query, queryRange)
-	if err != nil {
-		log.Errorf("QueryRange(%+v, %+v) failed with: %+v", req.Query, queryRange, err)
-		return nil, err
-	}
+		queryRange := v1.Range{
+			Start: start,
+			End:   end,
+			Step:  step,
+		}
 
-	if res.Type() != model.ValMatrix {
-		return nil, fmt.Errorf("Unexpected query result type: %s", res.Type())
+		res, err = s.prometheusAPI.QueryRange(ctx, req.Query, queryRange)
+		if err != nil {
+			log.Errorf("QueryRange(%+v, %+v) failed with: %+v", req.Query, queryRange, err)
+			return nil, err
+		}
+	} else {
+		res, err = s.prometheusAPI.Query(ctx, req.Query, time.Time{})
```

Comment:
This is "summary" mode?  Does that means this gets the single most recent datapoint or does it summarize over some range?

Reply:
Correct this is "summary" mode, which means return the single most recent data point. We pass in `time.Time{}` as a way to indicate we want the most recent data point:
https://prometheus.io/docs/prometheus/latest/querying/api/#instant-queries

We were previously doing this manually by querying the full timeseries, and then calling `filterQueryRsp` to filter to the latest data point.

For a bit more context, here is where the `timeseries` param in the `web` API translates into `Summarize` in the public API (perhaps as part of #312 we could make these terms more consistent):
https://github.com/runconduit/conduit/blob/2f114e69fad36c7eb40aefcbccf301a6c30470c6/web/srv/api_handlers.go#L140